spaceship/lib/spaceship/portal/portal_client.rb

```diff
@@ -532,7 +532,12 @@ def create_device!(device_name, device_id, mac: false)
         register: 'single'
       })
 
-      parse_response(req, 'devices').first
+      devices = parse_response(req, 'devices')
+      return devices.first unless devices.empty?
+
+      raise parse_response(req, 'validationMessages').map { |message|
```

Comment:
Can we make this raise an `UnexpectedResponse` like the following taken from https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/portal/portal_client.rb#L595...

```rb
raise UnexpectedResponse.new, "Couldn't download certificate, got this instead: #{a}"
```

Just want to be consistent with some other error messages 😊

Reply:
But... this is an expected response... which is why we could get the validation messages from the response. Though I added a case where the `validationMessages` array could be empty.